adapters/repos/db/vector/hnsw/commit_logger.go

```diff
@@ -418,14 +426,18 @@ func (l *hnswCommitLogger) startSwitchLogs() chan struct{} {
 	return cancelSwitchLog
 }
 
+func (l *hnswCommitLogger) PauseMaintenance() {
+	l.cancel <- struct{}{}
+}
```

Comment:
this is very similar to Shutdown method (except waiting for cancelComplete signal)
could shutdown be used instead in snapshot::PauseMaintenance?